src/services/blocks/BlocksService.spec.ts

```diff
@@ -178,13 +190,104 @@ describe('BlocksService', () => {
+				mockBlock789629,
+				that
 			);
 
 			expect(
 				calcFee?.calc_fee(BigInt(941325000000), 1247, BigInt(125000000))
 			).toBe('1257000075');
 		});
+
+		/**
+		 * Decorated should return undefined when on a polkadot or kusama
+		 * chain because the values are hardcoded to increase performance
+		 */
+		it('Should return undefined `decorated` for polkadot chain', async () => {
+			const { decorated } = await blocksService['createCalcFee'](
+				mockApi,
+				('0xParentHash' as unknown) as Hash,
+				mockBlock789629,
+				that
+			);
+
+			expect(decorated).toBe(undefined);
+		});
+
+		/**
+		 * Test decorated value against non polkadot kusama chain
+		 */
+		it('Should return a truthy decorated when running on a Non-Polkadot-Kusama chain', async () => {
+			(mockApi.runtimeVersion
+				.specName as unknown) = polkadotRegistry.createType('Text', 'westend');
+
+			const { decorated } = await blocksService['createCalcFee'](
+				mockApi,
+				('0xParentHash' as unknown) as Hash,
+				mockBlock789629,
+				that
+			);
+
+			/**
+			 * This checks to make sure decorated is not undefined as a return value
+			 * when the chain is Non-polkadot-kusama
+			 */
+			expect(decorated).toBeTruthy();
+
+			(mockApi.runtimeVersion
+				.specName as unknown) = polkadotRegistry.createType('Text', 'polkadot');
+		});
+
+		it('Should fill the cache with decorated and runtimeVersion data when running on a Non-Polkadot-Kusama chain', async () => {
+			(mockApi.runtimeVersion
+				.specName as unknown) = polkadotRegistry.createType('Text', 'westend');
+
+			await blocksService['createCalcFee'](
+				mockApi,
+				('0xParentHash' as unknown) as Hash,
+				mockBlock789629,
+				that
+			);
+
+			/**
+			 * This checks to make sure the cache is updated.
+			 */
+			expect(that.cache.decorated).toBeTruthy();
+			/**
+			 * Check runtimeVersion. Westend's actual version might be different
+			 * but because we are checking just createCalcFee we just need to make
+			 * sure it stores the mockApi block's default runtime.
+			 */
+			expect(that.cache.runtimeVersion).toBe(16);
+			(mockApi.runtimeVersion
+				.specName as unknown) = polkadotRegistry.createType('Text', 'polkadot');
+		});
+
+		it('Should cache the correct block runtime when different from the api runtime', async () => {
+			/**
+			 * The mockApi take runtimeVersion is shared between the api and block
+			 * so updating the mockApi runtimeVersion will update the version to be cached
+			 * from the block.
+			 */
+			(mockApi.runtimeVersion
+				.specVersion as unknown) = polkadotRegistry.createType('u32', 20);
+			(mockApi.runtimeVersion
+				.specName as unknown) = polkadotRegistry.createType('Text', 'westend');
+
+			await blocksService['createCalcFee'](
+				mockApi,
+				('0xParentHash' as unknown) as Hash,
+				mockBlock789629,
+				that
+			);
+
+			expect(that.cache.runtimeVersion).toBe(20);
+
+			(mockApi.runtimeVersion
+				.specVersion as unknown) = polkadotRegistry.createType('u32', 16);
```

Comment:
You could save these to temp variables at the beggining and then re-assign at the end instead of recreating

Reply:
f